include/sys_clock.h

```diff
@@ -28,6 +28,82 @@
 extern "C" {
 #endif
 
+/**
+ * @addtogroup clock_apis
+ * @{
+ */
+
```

Comment:
i'm missing here a common description of `k_timeout_t`. What it represents, what are the special values, etc. Recommendations (like using K_MSEC). This could be referenced by other API's